libevmasm/Assembly.cpp

```diff
@@ -74,6 +75,122 @@ unsigned Assembly::codeSize(unsigned subTagSize) const
 	}
 }
 
+void Assembly::addAssemblyItemsFromJSON(Json::Value const& _code)
+{
+	solAssert(m_items.empty());
+	solAssert(_code.isArray());
+	for (auto const& jsonItem: _code)
+		m_items.emplace_back(createAssemblyItemFromJSON(jsonItem));
+
+	for (auto current = m_items.begin(); current != m_items.end(); ++current)
+	{
+		// During the assembly json export a `JUMPDEST` is always generated after a `tag`.
+		// So we just ignore exactly these `JUMPDEST`'s.
+		auto const nextItem = next(current);
+		if (nextItem != m_items.end() &&
+			current->type() == AssemblyItemType::Tag &&
+			nextItem->type() == AssemblyItemType::Operation &&
+			nextItem->instruction() == Instruction::JUMPDEST
+		)
+			m_items.erase(nextItem);
+	}
+}
```

Comment:
You probably checked this: Are there ever any ``JUMPDEST``s left after removing the ones after tags above? Are there ever any tags that are not followed by a ``JUMPDEST``?
Just wondering whether we could or should just validate that there is exactly one jumpdest after each tag already when constructing ``m_items`` and validate that there are no other ``JUMPDEST``s, but there may be freak cases I'm not thinking about.

Reply:
I did not check this explicitly yet. But we could validate this of course. I could just add an additional check like this.